doc/source/scripts/base.py

```diff
@@ -25,11 +25,8 @@ class TableWriter(object):
     postprocess = []
     sort = True
 
-    def __getattr__(self, key):
-        try:
-            return self.fields[key]
-        except KeyError:
-            return super(TableWriter, self).__getattr__(key)
+    def __getattr__(self, key: str) -> list:
+        return self.fields[key]
 
     def __init__(self, *args, **kwargs):
         stem = os.getcwd().split("source")[0]
```

Comment:
(Non-blocking) this has always bothered me, I think starting from `__file__` is probably more robust, and using the `pathlib` library too -- but out of scope for this PR.